code/modules/vehicles/speedbike.dm

```diff
@@ -36,6 +36,362 @@
+		playsound(get_turf(owner),'sound/machines/buzz-two.ogg', 50, 1)
+		return
+	if(bike.CAN.air_contents.return_pressure() < ONE_ATMOSPHERE)
+		owner << "<span class='notice'>Alert: You have have exhausted your gas supply, refill or replace your canister.</span>"
+		playsound(get_turf(owner),'sound/machines/buzz-two.ogg', 50, 1)
+		return
+	if(!bike.CAN.valve_open)
+		owner << "<span class='notice'>Alert: You begin to dispense gas from the stored canister at [bike.CAN.release_pressure]kPa.</span>"
+		bike.CAN.valve_open = 1
+		button_icon_state = "flightpack_airbrake"
+		name = "Stop Gas Release"
+		UpdateButtonIcon()
+		return
+	if(bike.CAN.valve_open)
+		owner << "<span class='notice'>Alert: You have stopped dispensing gas from your stored canister, it has [bike.CAN.air_contents.return_pressure()]kPa of gas remaining.</span>"
+		bike.CAN.valve_open = 0
+		name = "Release Stored Gas"
+		button_icon_state = "flightpack_stabilizer"
+		UpdateButtonIcon()
+
+/datum/action/innate/atmos_bike/control
+	name = "eject canister"
+	desc = "Ejects the vehicle's internal canister"
+	button_icon_state = "mech_eject"
+	var/list/canisters_to_load = list()
+
+/datum/action/innate/atmos_bike/control/Activate()
+	if(!bike.CAN)
+		bike.loaded = FALSE
+	if(bike.loaded)
+		owner << "<span class='notice'>You eject the vehicle's internal canister.</span>"
+		bike.CAN.loc = get_turf(bike)
+		button_icon_state = "mech_cycle_equip_off"
+		name = "load canister"
+		desc = "Loads the vehicle's internal canister"
+		bike.loaded = FALSE
+		UpdateButtonIcon()
+		return
+	if(!bike.loaded)
+		for(var/obj/machinery/portable_atmospherics/canister/C in range(1,owner))
+			canisters_to_load += C
+		if(canisters_to_load.len==0)
+			owner << "<span class='notice'>There are no nearby canisters to load into the vehicle!</span>"
+			playsound(get_turf(owner),'sound/machines/buzz-two.ogg', 50, 1)
+			return
+		else
+			bike.CAN = input(owner, "Choose which canister to load", "Canisters:") as null|anything in canisters_to_load
+		if (!owner || QDELETED(owner) || !bike.CAN || QDELETED(bike.CAN))
+			return
+		bike.CAN.loc = bike
+		owner << "<span class='notice'>The vehicle scoops up the [bike.CAN] and locks it into position.</span>"
+		canisters_to_load.Cut()
+		bike.loaded = TRUE
+		name = "eject canister"
+		desc = "Ejects the vehicle's internal canister"
+		button_icon_state = "mech_eject"
+		UpdateButtonIcon()
+
+
+// Engineer's repair-bike
+
+/obj/vehicle/space/speedbike/repair
+	name = "prototype repair vehicle"
+	desc = "An experimental repair device mounted on a speederbike... what will they think of next."
+	icon_state = "engi_bike"
+	overlay_state = "cover_engi"
+	var/obj/machinery/repair_turret/turret = null
+
+/obj/vehicle/space/speedbike/repair/Initialize()
+	. = ..()
+	turret = new(loc)
+
+/obj/vehicle/space/speedbike/repair/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+	. = ..()
+	riding_datum = new/datum/riding/space/repair
```

Comment:
I'm going to call in @kevinz000 and @Militaires here because I'm almost certain this is incorrect